block_my_feedback.php

```diff
@@ -59,35 +61,220 @@ public function get_content(): stdClass {
         $this->content->footer = '';
 
         $template = new stdClass();
-        $template->feedback = $this->fetch_feedback($USER);
 
-        // Hide the block when no content.
-        if (!$template->feedback) {
-            return $this->content;
+        if (self::is_teacher()) {
+            // Teacher content.
+            $template->mods = $this->fetch_marking($USER);
+        } else {
+            // Student content.
+            $template->mods = $this->fetch_feedback($USER);
+            $template->showfeedbacktrackerlink = true;
+        }
+
+        if (isset($template->mods)) {
+            $this->content->text = $OUTPUT->render_from_template('block_my_feedback/content', $template);
         }
 
-        $this->content->text = $OUTPUT->render_from_template('block_my_feedback/content', $template);
         return $this->content;
     }
 
+    /**
+     * Return if user has archetype editingteacher.
+     *
+     */
+    public static function is_teacher(): bool {
+        global $DB, $USER;
+        // Get id's from role where archetype is editingteacher.
+        $roles = $DB->get_fieldset('role', 'id', ['archetype' => 'editingteacher']);
+
+        // Check if user has editingteacher role on any courses.
+        list($roles, $params) = $DB->get_in_or_equal($roles, SQL_PARAMS_NAMED);
+        $params['userid'] = $USER->id;
+        $sql = "SELECT id
+                FROM {role_assignments}
+                WHERE userid = :userid
+                AND roleid $roles";
+        return  $DB->record_exists_sql($sql, $params);
+    }
+
+    /**
+     * Return marking for a user.
+     *
+     * @param stdClass $user
+     */
+    public function fetch_marking(stdClass $user): ?array {
+        // User courses.
+        $courses = enrol_get_all_users_courses($user->id, false, ['enddate']);
+        // Marking.
+        $marking = [];
+
+        foreach ($courses as $course) {
+            // Skip hidden courses.
+            if (!$course->visible) {
+                continue;
+            }
+            // Skip none current course.
+            if (!self::is_course_current($course)) {
```

Comment:
`is_course_current` is not a static function, call it by using `$this` or change the method to static?